website/project/model.py

```diff
@@ -1128,6 +1146,36 @@ def can_comment(self, auth):
             )
         return self.is_contributor(auth.user)
 
+    def set_node_license(self, data, auth, save=True):
```

Comment:
It might make sense to check the permissions of `auth` in this method, as other Node methods do.